bundles/statistics/statsgrid2016/components/IndicatorParameters.js

```diff
@@ -70,7 +73,14 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.IndicatorParameters', function
          me.spinner.insertTo(this.parentElement.parent());
          me.spinner.start();
         //get the data to create ui with
-        me.paramHandler.getData( datasrc, indId, elements );
+        me.paramHandler.getData( datasrc, indId, regionsetRestriction, elements );
+    },
+    refresh: function ( datasrc, indId, regionsetRestriction ) {
+        this.clean();
+        // only refresh data if component has been initialized
+        if ( this._initialized ) {
```

Comment:
RegionSelector is created on indicatorSelected() as well. Would it be enough to check if it has been created instead of using another variable just for this? Or why is this check needed? What's the difference between indicatorSelected() and refresh()?

Reply:
Maybe you can get rid of refresh() if you just skip the spinner lines on indicatorSelected() when the spinner is already shown on the UI?